lib/fast_jsonapi/serialization_core.rb

```diff
@@ -26,9 +26,10 @@ def ids_hash(ids, record_type)
         id_hash(ids, record_type) # ids variable is just a single id here
       end
 
-      def id_hash_from_record(record, record_types = nil)
-        record_type = record_types.respond_to?(:key) ? record_types[record.class] : nil
-        { id: record.id.to_s, type: (record_type || record.class.name.underscore.to_sym) }
+      def id_hash_from_record(record, record_types)
+        # memoize the record type within the record_types dictionary, then assigning to record_type:
+        record_type = record_types[record.class] ||= record.class.name.underscore.to_sym
```

Comment:
@christophersansone Not yet a method like you suggested (that's something I can refactor, tho), but caching like a boss...